core/src/main/java/org/infinispan/persistence/spi/CacheWriter.java

```diff
@@ -35,4 +38,28 @@
     * @throws PersistenceException in case of an error, e.g. communicating with the external storage
     */
    boolean delete(Object key);
+
+   /**
+    * Persist all provided entries to the store in a single batch update. If this is not supported by the
+    * underlying store, then entries are written to the store sequentially via {@link #write(MarshalledEntry)}
```

Comment:
Ah, you're right that async write already does the coalescing. In that case I think that Collection would be more appropriate. Documentation still needs these details, though.

However, it seems you've missed one use-case that would benefit from this: `CacheWriterInterceptor$Updater`. If the store does not support `TransactionalCacheWriter`, the update should be still batched.